packages/sdk/src/vdp/artifact/ArtifactClient.ts

```diff
@@ -0,0 +1,245 @@
+import { getQueryString } from "../../helper";
+import { APIResource } from "../../main/resource";
+import {
+  Chunk,
+  CreateKnowledgeBaseRequest,
+  DeleteKnowledgeBaseFileRequest,
+  DeleteKnowledgeBaseRequest,
+  File,
+  GetChunkContentRequest,
+  GetFileContentRequest,
+  GetFileDetailsRequest,
+  GetSourceFileRequest,
+  KnowledgeBase,
+  ListChunksRequest,
+  ListChunksResponse,
+  ListKnowledgeBaseFilesRequest,
+  ListKnowledgeBaseFilesResponse,
+  ListKnowledgeBasesRequest,
+  ListKnowledgeBasesResponse,
+  ProcessKnowledgeBaseFilesRequest,
+  SimilarityChunk,
+  SimilarityChunksSearchRequest,
+  SourceFile,
+  UpdateChunkRequest,
+  UpdateKnowledgeBaseRequest,
+  UploadKnowledgeBaseFileRequest,
+} from "./types";
+
+export class ArtifactClient extends APIResource {
+  async listKnowledgeBases(
+    props: ListKnowledgeBasesRequest,
+  ): Promise<ListKnowledgeBasesResponse> {
+    const { ownerId, pageSize, pageToken } = props;
+    try {
+      const queryString = getQueryString({
+        baseURL: `/namespaces/${ownerId}/knowledge-bases`,
+        pageSize,
+        pageToken,
+      });
+      const data =
+        await this._client.get<ListKnowledgeBasesResponse>(queryString);
+      return Promise.resolve(data);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  async createKnowledgeBase(
+    props: CreateKnowledgeBaseRequest,
+  ): Promise<KnowledgeBase> {
+    const { ownerId, payload } = props;
+    try {
+      const data = await this._client.post<{ knowledge_base: KnowledgeBase }>(
```

Comment:
@thewbuk Does backend still use snake_case here?

And please put this type as CreateKnowledgeBaseResponse